controllers/feeds/getFeedsV2.js

```diff
@@ -254,10 +254,45 @@ module.exports = async (req, res) => {
       }
     }
     // get karma score for each post actor
+    for (let i = 0; i < data.length; i++) {
+      if (data[i].latest_reactions?.downvotes) {
+        for (let j = 0; j < data[i].latest_reactions?.downvotes.length; j++) {
+          if (!postActors.includes(data[i].latest_reactions.downvotes[j].user_id)) {
+            postActors.push(data[i].latest_reactions.downvotes[j].user_id);
+          }
+        }
+      }
+      if (data[i].own_reactions?.downvotes) {
+        for (let j = 0; j < data[i].own_reactions?.downvotes.length; j++) {
+          if (!postActors.includes(data[i].own_reactions.downvotes[j].user_id)) {
+            postActors.push(data[i].own_reactions.downvotes[j].user_id);
+          }
+        }
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The current approach of checking if a user_id is already in the `postActors` array before pushing it has a time complexity of O(n^2) which can be improved. Instead, consider using a Set to store unique user_ids. This will reduce the time complexity to O(n) as lookup, insertion and deletion operations in a Set are O(1).
```diff
- let postActors = [];
+ let postActors = new Set();

// ...

- if (!postActors.includes(data[i].latest_reactions.downvotes[j].user_id)) {
-   postActors.push(data[i].latest_reactions.downvotes[j].user_id);
- }
+ postActors.add(data[i].latest_reactions.downvotes[j].user_id);

// ...

- if (!postActors.includes(data[i].own_reactions.downvotes[j].user_id)) {
-   postActors.push(data[i].own_reactions.downvotes[j].user_id);
- }
+ postActors.add(data[i].own_reactions.downvotes[j].user_id);

// Convert the set back to an array when passing it to getUsersKarmaScore function
- const karmaScores = await UsersFunction.getUsersKarmaScore(User, postActors);
+ const karmaScores = await UsersFunction.getUsersKarmaScore(User, Array.from(postActors));
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->